src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -184,22 +198,17 @@ class AuthScreens extends React.Component {
     }
 
     /**
-     * @param {String} [url]
-     * @returns {Boolean}
+     * Navigate to the transition exit route
+     *
+     * @param {String} exitTo
      */
-    shouldCreateFreePolicy(url = '') {
-        if (!url) {
-            return false;
-        }
-
-        const path = new URL(url).pathname;
-        const params = new URLSearchParams(url);
-        const exitTo = params.get('exitTo');
-        const email = params.get('email');
-        const isLoggingInAsNewUser = !_.isNull(this.props.session.email) && (email !== this.props.session.email);
-        return !isLoggingInAsNewUser
-            && Str.startsWith(path, Str.normalizeUrl(ROUTES.LOGIN_WITH_SHORT_LIVED_TOKEN))
-            && exitTo === ROUTES.WORKSPACE_NEW;
+    navigateToExitRoute(exitTo) {
+        // In order to navigate to a modal, we first have to dismiss the current modal. Without dismissing the current modal, if the user cancels out of the workspace modal,
+        // then they will be routed back to /transition/<accountID>/<email>/<authToken>/workspace/<policyID>/card and we don't want that. We want them to go back to `/`
+        // and by calling dismissModal(), the /transition/... route is removed from history so the user will get taken to `/` if they cancel out of the new workspace modal.
```

Comment:
**idea:** this comment can be simplified to just say...

>We must call dismissModal() to remove the /transition route from history

The rest of it is confusing and very specific to the "workspace modal". but we are now generically navigating to an `exitTo` here.

Reply:
Yeah good idea haha.